frontend/public/components/factory/details.tsx

```diff
@@ -88,11 +90,19 @@ export const DetailsPage = withFallback<DetailsPageProps>(({ pages = [], ...prop
           component: (cProps) => renderAsyncComponent(p, cProps),
         })),
       ...dynamicResourcePageExtensions
-        .filter(
-          (p) =>
-            referenceForExtensionModel(p.properties.model) ===
-            (kindObj ? referenceFor(kindObj) : props.kind),
-        )
+        .filter((p) => {
+          if (p.properties.model.version) {
+            return (
+              referenceForExtensionModel(p.properties.model as ExtensionK8sModel) ===
+              (kindObj ? referenceFor(kindObj) : props.kind)
+            );
+          }
+          return (
+            referenceForModel(
+              findDiscoveryModelForGroupKind(p.properties.model.group, p.properties.model.kind),
+            ) === (kindObj ? referenceFor(kindObj) : props.kind)
+          );
```

Comment:
Using `findDiscoveryModelForGroupKind` is incorrect here because a lookup by `group` and `kind` can result in picking up the wrong modal with the preferred version instead of the one which matches `kindObj`.

What you really need to do is simply check if the `kindObj` has the same `group` and `kind` of the extension.